### ISSUE TITLE
Saving the model with state dict BEFORE training

### ISSUE THREAD

**Issue by pfuerste** [open-mmlab/mmdetection#6963]
Hi, I am trying to save a model before training, because I want to check the change of gradients in certain layers. Sadly, I cant manage to save it in the same way as it is done by the hooks apparently: No matter if I save with torch.save(model. path) or runner._save_checkpoint(), the size of the model (in MB) differs vastly. Is there a way to save the whole model with weights before training?

Also, I always get "missing keys in source state_dict" and "size mismatch for bbox_head.cls_branches.0.weight" when loading a model if I try to load_from and also change the backbone. Am I making mistakes in defining the model? This is my config:

```
_base_ = '/home/fuerste/mmdetection/configs/deformable_detr/deformable_detr_r50_16x2_50e_coco.py'

model = dict(
    backbone=dict(
        type='ResNet',
        depth=101,
        num_stages=4,
        out_indices=(1, 2, 3),
        frozen_stages=1,
        norm_cfg=dict(type='BN', requires_grad=False),
        norm_eval=True,
        style='pytorch',
        init_cfg=dict(type='Pretrained',
                      # checkpoint="https://download.openmmlab.com/mmclassification/v0/resnet/resnet101_8xb32_in1k_20210831-539c63f8.pth")),
                      checkpoint='torchvision://resnet101')),
    neck=dict(
        type='ChannelMapper',
        in_channels=[512, 1024, 2048],
        kernel_size=1,
        out_channels=256,
        act_cfg=None,
        norm_cfg=dict(type='GN', num_groups=32),
        num_outs=4),
    bbox_head=dict(
        type='DeformableDETRHead',
        num_classes=17))


dataset_type = 'COCODataset'
classes = (
    ('bobcat', 'opossum', 'empty', 'coyote', 'raccoon', 'bird',
     'dog', 'cat', 'squirrel', 'rabbit', 'skunk', 'rodent',
     'badger', 'deer', 'car', 'fox', 'animal')
)

img_scale = (1024, 600)
img_norm_cfg = dict(
    mean=[123.675, 116.28, 103.53], std=[58.395, 57.12, 57.375], to_rgb=True)
train_pipeline = [
    dict(type='LoadImageFromFile'),
    dict(type='LoadAnnotations', with_bbox=True),
    dict(type='Resize', img_scale=img_scale, keep_ratio=True),
    dict(type='RandomFlip', flip_ratio=0.5),
    dict(type='Normalize', **img_norm_cfg),
    dict(type='Pad', size_divisor=32),
    dict(type='DefaultFormatBundle'),
    dict(type='Collect', keys=['img', 'gt_bboxes', 'gt_labels']),
]
test_pipeline = [
    dict(type='LoadImageFromFile'),
    dict(
        type='MultiScaleFlipAug',
        img_scale=img_scale,
        flip=False,
        transforms=[
            dict(type='Resize', keep_ratio=True),
            dict(type='RandomFlip'),
            dict(type='Normalize', **img_norm_cfg),
            dict(type='Pad', size_divisor=32),
            dict(type='ImageToTensor', keys=['img']),
            dict(type='Collect', keys=['img']),
        ])
]

load_from = '/home/fuerste/mmdetection/checkpoints/deformable_detr_r50_16x2_50e_coco_20210419_220030-a12b9512.pth'

samples_per_gpu = 2
workers_per_gpu = 1

data = dict(
    samples_per_gpu=samples_per_gpu,
    workers_per_gpu=workers_per_gpu,
    train=dict(
        img_prefix='/home/datasets/camera_traps/Caltech-Camera-Traps/CCT20-benchmark/eccv_18_all_images_sm',
        classes=classes,
        filter_empty_gt=False,
        ann_file='/home/fuerste/thesis_root/data/cct20/annotations/hybrid/_5_train_annotations.json',
        pipeline=train_pipeline),
    val=dict(
        img_prefix='/home/datasets/camera_traps/Caltech-Camera-Traps/CCT20-benchmark/eccv_18_all_images_sm',
        classes=classes,
        filter_empty_gt=False,
        separate_eval=True,
        ann_file=['/home/fuerste/thesis_root/data/cct20/annotations/hybrid/_5_cis_val_annotations.json',
                  '/home/fuerste/thesis_root/data/cct20/annotations/hybrid/_5_trans_val_annotations.json'],
        pipeline=test_pipeline),
    test=dict(
        img_prefix='/home/datasets/camera_traps/Caltech-Camera-Traps/CCT20-benchmark/eccv_18_all_images_sm',
        classes=classes,
        filter_empty_gt=False,
        separate_eval=True,
        ann_file=['/home/fuerste/thesis_root/data/cct20/annotations/hybrid/_5_cis_test_annotations.json',
                  '/home/fuerste/thesis_root/data/cct20/annotations/hybrid/_5_trans_test_annotations.json'],
        pipeline=test_pipeline))

log_config = dict(
    interval=10,
    hooks=[
        dict(type='TextLoggerHook'),
        dict(
            type='WandbLoggerHook',
            init_kwargs=dict(
                project='hybrid testing',
                config={},
                tags=["cct20", "hybrid naive"]
            ))])


workflow = [('train', 1), ('val', 1)]
runner = dict(type='EpochBasedRunner', max_epochs=1)
optimizer = dict(
    type='AdamW',
    lr=2e-5,
    weight_decay=0.0001,
    paramwise_cfg=dict(
        custom_keys={
            'backbone': dict(lr_mult=0.5),
            'sampling_offsets': dict(lr_mult=0.1),
            'reference_points': dict(lr_mult=0.1)
        }))

optimizer_config = dict(grad_clip=dict(max_norm=0.1, norm_type=2))

lr_config = dict(
    policy='step',
    step=[25],
    warmup='linear',
    warmup_iters=500,
    warmup_ratio=0.001)

```
